apps/condo/domains/property/components/panels/MapConstructor.spec.ts

```diff
@@ -0,0 +1,145 @@
+            })
+            it('after adding unit with not-numeric label shouldn\'t update names of other units', () => {
+                expect(map.sections[0].floors[3].units[2].label).toEqual(unitPlacedBefore.label)
+            })
+        })
+
+        describe('Edit unit', () => {
+            let map = mapAddSection(null, { id: null, minFloor: -2, maxFloor: 5, unitsOnFloor: 10 })
+            map = mapAddSection(map, { id: null, minFloor: -1, maxFloor: 10, unitsOnFloor: 10 })
+            map = mapAddSection(map, { id: null, minFloor: 2, maxFloor: 7, unitsOnFloor: 4 })
+            const unitPlacedBefore = map.sections[0].floors[3].units[2]
+            const unitPlacedAfter = map.sections[2].floors[4].units[1]
+            const editUnit = map.sections[1].floors[2].units[0]
+            it('should correctly update names of some units if has numeric label', () => {
+                map = mapUpdateUnit(map, { ...editUnit, label: '1000', floor: map.sections[1].floors[2].id, section: map.sections[1].id } )
+                expect(map.sections[2].floors[4].units[1].label).not.toEqual(unitPlacedAfter.label)
+                expect(map.sections[0].floors[3].units[2].label).toEqual(unitPlacedBefore.label)
+            })
+            it('should correctly update map if floor/section is changed', () => {
+                map = mapUpdateUnit(map, { ...editUnit, floor: map.sections[0].floors[0].id, section: map.sections[0].id } )
+                const info = getUnitInfo(map, editUnit.id)
+                expect(info.floor).toEqual(map.sections[0].floors[0].id)
+                expect(info.section).toEqual(map.sections[0].id)
+            })
+        })
+
+        describe('Delete unit', () => {
+            it('should remove unit from map', () => {
+                let map = mapAddSection(null, { id: null, minFloor: -2, maxFloor: 5, unitsOnFloor: 10 })
+                const unitToRemove = map.sections[0].floors[1].units[2]
+                map = mapRemoveUnit(map, unitToRemove.id)
+                const found = getUnitInfo(map, unitToRemove.id)
+                expect(found.floor).toBe('')
+            })
+        })
+    })
+        
+    describe('Section operations', () => {
+        describe('Add section', () => {
+            it('check generated structure on section add', () => {
+                let map = mapAddSection(null, { id: null, minFloor: -5, maxFloor: 20, unitsOnFloor: 10 })
+                map = mapAddSection(map, { id: null, minFloor: -10, maxFloor: 10, unitsOnFloor: 10 })
+                expect(map.sections).toHaveLength(2)
+                expect(map.sections[1].floors).toHaveLength(20)
+            })
+        })
+        describe('Edit section', () => {
+            it('check rename of the structure on section edit', () => {
+                let map = mapAddSection(null, { id: null, minFloor: -5, maxFloor: 20, unitsOnFloor: 10, name: 'test_name_1' })
```

Comment:
It's hard to understand the map structure. We should have tests for the structure